src/webgpu/api/validation/queue/copyToTexture/CopyExternalImageToTexture.spec.ts

```diff
@@ -290,40 +295,75 @@ g.test('source_image,crossOrigin')
       .combine('sourceImage', ['canvas', 'offscreenCanvas', 'imageBitmap'])
       .combine('isOriginClean', [true, false])
       .beginSubcases()
+      .combine('contentFrom', ['image', 'imageBitmap', 'canvas', 'offscreenCanvas'] as const)
       .combine('copySize', [
         { width: 0, height: 0, depthOrArrayLayers: 0 },
         { width: 1, height: 1, depthOrArrayLayers: 1 },
       ])
   )
   .fn(async t => {
-    const { sourceImage, isOriginClean, copySize } = t.params;
+    const { sourceImage, isOriginClean, contentFrom, copySize } = t.params;
 
     const crossOriginUrl = 'https://get.webgl.org/conformance-resources/opengl_logo.jpg';
     const originCleanUrl = '../out/resources/Di-3d.png';
 
     const img = document.createElement('img');
     img.src = isOriginClean ? originCleanUrl : crossOriginUrl;
+
+    // Load image
+    const timeout_ms = 5000;
     try {
-      await awaitOrTimeout(img.decode());
+      await raceWithRejectOnTimeout(img.decode(), timeout_ms, 'load image timeout');
     } catch (e) {
-      t.skip('Cannot load image');
+      if (isOriginClean) {
+        throw e;
+      } else {
+        t.warn('Something wrong happens in get.webgl.org');
+        t.skip('Cannot load image in time');
```

Comment:
I guess we should leave a TODO in the description to try to make this test run without accessing the internet. WPT has provisions for this:
https://web-platform-tests.org/writing-tests/server-features.html#tests-involving-multiple-origins

However our test suite (and local test server) are not set up to be able to use them.